extras/exporters/gltf-exporter.js

```diff
@@ -0,0 +1,465 @@
+const ARRAY_BUFFER = 34962;
+const ELEMENT_ARRAY_BUFFER = 34963;
+
+const getIndexComponentType = (indexFormat) => {
+    switch (indexFormat) {
+        case pc.INDEXFORMAT_UINT8: return 5121;
+        case pc.INDEXFORMAT_UINT16: return 5123;
+        case pc.INDEXFORMAT_UINT32: return 5125;
+    }
+    return 0;
+};
+
+const getComponentType = (dataType) => {
+    switch (dataType) {
+        case pc.TYPE_INT8: return 5120;
+        case pc.TYPE_UINT8: return 5121;
+        case pc.TYPE_INT16: return 5122;
+        case pc.TYPE_UINT16: return 5123;
+        case pc.TYPE_INT32: return 5124;
+        case pc.TYPE_UINT32: return 5125;
+        case pc.TYPE_FLOAT32: return 5126;
+    }
+    return 0;
+};
+
+const getAccessorType = (componentCount) => {
+    switch (componentCount) {
+        case 1: return 'SCALAR';
+        case 2: return 'VEC2';
+        case 3: return 'VEC3';
+        case 4: return 'VEC4';
+    }
+    return 0;
+};
+
+const getSemantic = (engineSemantic) => {
+    switch (engineSemantic) {
+        case pc.SEMANTIC_POSITION: return 'POSITION';
+        case pc.SEMANTIC_NORMAL: return 'NORMAL';
+        case pc.SEMANTIC_TANGENT: return 'TANGENT';
+        case pc.SEMANTIC_COLOR: return 'COLOR_0';
+        case pc.SEMANTIC_BLENDINDICES: return 'JOINTS_0';
+        case pc.SEMANTIC_BLENDWEIGHT: return 'WEIGHTS_0';
+        case pc.SEMANTIC_TEXCOORD0: return 'TEXCOORD_0';
+        case pc.SEMANTIC_TEXCOORD1: return 'TEXCOORD_1';
+        case pc.SEMANTIC_TEXCOORD2: return 'TEXCOORD_2';
+        case pc.SEMANTIC_TEXCOORD3: return 'TEXCOORD_3';
+        case pc.SEMANTIC_TEXCOORD4: return 'TEXCOORD_4';
+        case pc.SEMANTIC_TEXCOORD5: return 'TEXCOORD_5';
+        case pc.SEMANTIC_TEXCOORD6: return 'TEXCOORD_6';
+        case pc.SEMANTIC_TEXCOORD7: return 'TEXCOORD_7';
+    }
+};
```

Comment:
In `glb-parser.js` there is:

```js
const gltfToEngineSemanticMap = {
    'POSITION': SEMANTIC_POSITION,
    'NORMAL': SEMANTIC_NORMAL,
    'TANGENT': SEMANTIC_TANGENT,
    'COLOR_0': SEMANTIC_COLOR,
    'JOINTS_0': SEMANTIC_BLENDINDICES,
    'WEIGHTS_0': SEMANTIC_BLENDWEIGHT,
    'TEXCOORD_0': SEMANTIC_TEXCOORD0,
    'TEXCOORD_1': SEMANTIC_TEXCOORD1,
    'TEXCOORD_2': SEMANTIC_TEXCOORD2,
    'TEXCOORD_3': SEMANTIC_TEXCOORD3,
    'TEXCOORD_4': SEMANTIC_TEXCOORD4,
    'TEXCOORD_5': SEMANTIC_TEXCOORD5,
    'TEXCOORD_6': SEMANTIC_TEXCOORD6,
    'TEXCOORD_7': SEMANTIC_TEXCOORD7
};
```

I think an object in itself is shorter, but since its basically the same object (just inverted), I started to think these variables could be exported and then just automatically inverted, for example:

```js
function objectInvert(obj) {
    const ret = {};
    for (const key in obj) {
        const val = obj[key];
        ret[val] = key;
    }
    return ret;
}
export const engineToGltfSemanticMap = objectInvert(gltfToEngineSemanticMap);
```
(but maybe this is just overcomplicating things)

Reply:
I started with the object initially .. but the engine needs to be loaded before the pcx (extras) .. and so a static time access to pc constants is not possible here - and so I added a function instead, which executes after the engine has been loaded.
When we have full tree-shaking, and make this a proper module, we'd definitely do something else.